javascript-view-swapping/main.js

```diff
@@ -0,0 +1,27 @@
+
+var $tabContainer = document.querySelector('.tab-container');
+var $tabList = document.querySelectorAll('.tab');
+var $viewList = document.querySelectorAll('.view');
+
+$tabContainer.addEventListener('click', function (e) {
+  var isTab = e.target.matches('.tab');
+  var $currentDataView = e.target.getAttribute('data-view');
+
+  if (isTab) {
+    for (var i = 0; i < $tabList.length; i++) {
+      if (e.target !== $tabList[i]) {
+        $tabList[i].className = 'tab';
+      } else {
+        e.target.className = 'tab active';
+      }
+    }
+  }
+
+  for (var j = 0; j < $viewList.length; j++) {
```

Comment:
You will want this code in your conditional as well. As it is now, this code will run whether the event target element has the `tab` class or not, which is not the intended functionality.